sponsor-dapp/src/components/ContractInteraction.js

```diff
@@ -47,6 +49,36 @@ class ContractInteraction extends Component {
     );
   }
 
+  calcMaxTokensThatCanBeCreated() {
+    const { drizzleHelper } = this;
+    const { contractAddress } = this.props;
+    const { web3 } = this.props.drizzle;
+
+    const fpScalingFactor = BigNumber(web3.utils.toWei("1", "ether"));
+    const newExcessMargin = BigNumber(drizzleHelper.getCache(contractAddress, "calcExcessMargin", []));
+    const newUnderlyingPrice = BigNumber(
+      drizzleHelper.getCache(contractAddress, "getUpdatedUnderlyingPrice", []).underlyingPrice
+    );
+
+    const derivativeStorage = drizzleHelper.getCache(contractAddress, "derivativeStorage", []);
+    const startingUnderlyingPrice = BigNumber(derivativeStorage.referenceTokenState.underlyingPrice);
+    const startingTokenPrice = BigNumber(derivativeStorage.referenceTokenState.tokenPrice);
+    const supportedMove = BigNumber(derivativeStorage.fixedParameters.supportedMove);
+    const leverage = BigNumber(
+      drizzleHelper.getCache(derivativeStorage.externalAddresses.returnCalculator, "leverage", [])
+    );
+
+    const maxTokens = newExcessMargin
+      .times(startingUnderlyingPrice)
+      .times(fpScalingFactor)
+      .div(startingTokenPrice)
+      .div(newUnderlyingPrice)
+      .div(supportedMove)
```

Comment:
We aren't doing intermediate rounding here like we do in solidity, so results may slightly differ. It might warrant a comment to note that that level of precision shouldn't matter because we're rounding to 4 decimal points.

Reply:
Done.